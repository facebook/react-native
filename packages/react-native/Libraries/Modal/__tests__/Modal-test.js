/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

const render = require('../../../jest/renderer');
const View = require('../../Components/View/View');
const Modal = require('../Modal');
const React = require('react');

describe('Modal', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('should render as <Modal> when mocked', async () => {
    const instance = await render.create(
      <Modal>
        <View />
      </Modal>,
    );
    expect(instance).toMatchSnapshot();
  });

  it('should not render <Modal> when mocked with visible=false', async () => {
    const instance = await render.create(
      <Modal visible={false}>
        <View testID="child" />
      </Modal>,
    );
    expect(instance.toJSON()).toBeNull();
  });

  it('should render as <RCTModalHostView> when not mocked', async () => {
    jest.dontMock('../Modal');

    const instance = await render.create(
      <Modal>
        <View />
      </Modal>,
    );
    expect(instance).toMatchSnapshot();
  });

  it('should not render <RCTModalHostView> when not mocked with visible=false', async () => {
    jest.dontMock('../Modal');

    const instance = await render.create(
      <Modal visible={false}>
        <View />
      </Modal>,
    );
    expect(instance.toJSON()).toBeNull();
  });
});
