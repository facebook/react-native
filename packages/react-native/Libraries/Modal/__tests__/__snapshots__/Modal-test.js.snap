// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render as <Modal> when mocked 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View />
</Modal>
`;

exports[`Modal should render as <RCTModalHostView> when not mocked 1`] = `
<RCTModalHostView
  animationType="none"
  hardwareAccelerated={false}
  identifier={0}
  onDismiss={[Function]}
  onStartShouldSetResponder={[Function]}
  presentationStyle="fullScreen"
  style={
    Object {
      "position": "absolute",
    }
  }
  visible={true}
>
  <View
    collapsable={false}
    style={
      Array [
        Object {
          "flex": 1,
          "left": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={true}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View />
      </View>
    </View>
  </View>
</RCTModalHostView>
`;
